### PR TITLE
Simplify the simplification code.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/Workspaces/Core/Portable/Simplification/AbstractSimplificationService.cs [hunk 1]
```diff
@@ -328,20 +331,4 @@ protected virtual SyntaxNode TransformReducedNode(SyntaxNode reducedNode, Syntax
 
         protected abstract void GetUnusedNamespaceImports(SemanticModel model, HashSet<SyntaxNode> namespaceImports, CancellationToken cancellationToken);
     }
-
-    internal struct NodeOrTokenToReduce
-    {
-        public readonly SyntaxNodeOrToken NodeOrToken;
-        public readonly bool SimplifyAllDescendants;
-        public readonly SyntaxNodeOrToken OriginalNodeOrToken;
-        public readonly bool CanBeSpeculated;
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Simplification/AbstractSimplificationService.cs:337]: canbespeculated was never used.  removed.

File: src/Workspaces/CSharp/Portable/Simplification/CSharpSimplificationService.NodesAndTokensToReduceComputer.cs [hunk 2]
```diff
@@ -21,9 +21,6 @@ private class NodesAndTokensToReduceComputer : CSharpSyntaxRewriter
             private readonly List<NodeOrTokenToReduce> _nodesAndTokensToReduce;
             private readonly Func<SyntaxNodeOrToken, bool> _isNodeOrTokenOutsideSimplifySpans;
 
-            private static readonly Func<SyntaxNode, bool> s_containsAnnotations = n => n.ContainsAnnotations;
-            private static readonly Func<SyntaxNodeOrToken, bool> s_hasSimplifierAnnotation = n => n.HasAnnotation(Simplifier.Annotation);
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/Simplification/CSharpSimplificationService.NodesAndTokensToReduceComputer.cs:25]: moved up and shared between c# and vb.

File: src/Workspaces/Core/Portable/Simplification/AbstractSimplificationService.cs [hunk 3]
```diff
@@ -30,7 +30,10 @@ internal abstract class AbstractSimplificationService<TExpressionSyntax, TStatem
         where TStatementSyntax : SyntaxNode
         where TCrefSyntax : SyntaxNode
     {
-        private ImmutableArray<AbstractReducer> _reducers;
+        protected static readonly Func<SyntaxNode, bool> s_containsAnnotations = n => n.ContainsAnnotations;
+        protected static readonly Func<SyntaxNodeOrToken, bool> s_hasSimplifierAnnotation = n => n.HasAnnotation(Simplifier.Annotation);
+
+        private readonly ImmutableArray<AbstractReducer> _reducers;
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Simplification/AbstractSimplificationService.cs:36]: made readonly.

File: src/Workspaces/VisualBasic/Portable/Simplification/VisualBasicSimplificationService.NodesAndTokensToReduceComputer.vb [hunk 4]
```diff
@@ -124,39 +121,31 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Simplification
             End Function
 
             Public Overrides Function VisitMethodBlock(node As MethodBlockSyntax) As SyntaxNode
-                Dim updateFunc As Func(Of MethodBlockBaseSyntax, MethodBaseSyntax, SyntaxList(Of StatementSyntax), EndBlockStatementSyntax, MethodBlockBaseSyntax) =
+                Return VisitMethodBlockBaseSyntax(node,
                     Function(n, b, s, e)
                         Return DirectCast(n, MethodBlockSyntax).Update(node.Kind, DirectCast(b, MethodStatementSyntax), s, e)
-                    End Function
```
Comment (CyrusNajmabadi) [inline src/Workspaces/VisualBasic/Portable/Simplification/VisualBasicSimplificationService.NodesAndTokensToReduceComputer.vb:130]: inlined lambdas.

**Review by akhera99 — APPROVED** [review]

**Pull request merged**